Python/ceval.c

```diff
@@ -6626,7 +6628,7 @@ unicode_concatenate(PyThreadState *tstate, PyObject *v, PyObject *w,
             PyObject *names = _PyFrame_GetCode(f)->co_names;
             PyObject *name = GETITEM(names, oparg);
             if (name == NULL) {
-                name = _PyHydrate_LoadName(PyFrame_GetCode(f)->co_pyc, oparg);
+                name = _PyHydrate_LoadName(PyFrame_GetCode(f)->co_pyc, PyFrame_GetCode(f)->co_strings_start + oparg);
```

Comment:
If you go make a macro, I would not try to make the macro so smart that it can handle this case as well. 13 out of 14 is good enough.